### PR TITLE
add csrf protection for resources available to everyone

### PR DESCRIPTION
**Pull request by erlenner**
Reservations made through sulten were previously vulnerable to csrf attacks.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -14,6 +14,16 @@ class ApplicationController < ActionController::Base
   # Helper methods that are also used in controllers
   helper_method :current_user, :logged_in?
 
+  def handle_unverified_request
+    raise ActionController::InvalidAuthenticityToken
+  end
+
+  rescue_from ActionController::InvalidAuthenticityToken do |exception|
+    reset_session
+    flash[:error] = "The csrf token for your session was wrong"
```
Comment (cristeahub) [inline app/controllers/application_controller.rb]: This message is very developer-centric. The users of this system doesn't know what a csrf token is, so I would suggest having a simpler error message, like: "There was a problem processing your request. Please try again later.".

Let's not have unfortunate people get confused 😂

image

**Comment by FluenesHerre** [comment]
This manually implements the default behaviour in rails 4.
[https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L194](url)

**Comment by erlenner** [comment]
This manually implements the behaviour in [rails 4](https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L194)

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -14,6 +14,16 @@ class ApplicationController < ActionController::Base
   # Helper methods that are also used in controllers
   helper_method :current_user, :logged_in?
 
+  def handle_unverified_request
+    raise ActionController::InvalidAuthenticityToken
+  end
+
+  rescue_from ActionController::InvalidAuthenticityToken do |exception|
```
Comment (cristeahub) [inline app/controllers/application_controller.rb]: You never use the `exception` part here. I guess it can be ommited.

**Pull request merged**